src/pudl/etl/check_foreign_keys.py

```diff
@@ -50,15 +61,154 @@ def pudl_check_fks(logfile: pathlib.Path, loglevel: str):
     # Display logged output from the PUDL package:
     pudl.logging_helpers.configure_root_logger(logfile=logfile, loglevel=loglevel)
 
-    context = build_init_resource_context()
-    io_manager = pudl_sqlite_io_manager(context)
+    # Using PudlPaths to get default value for CLI causes validation issues
+    if not db_path:
+        db_path = PudlPaths().output_dir / "pudl.sqlite"
 
-    database_path = io_manager.base_dir / f"{io_manager.db_name}.sqlite"
-    logger.info(f"Checking foreign key constraints in {database_path}")
+    engine = sa.create_engine(f"sqlite:///{db_path}")
 
-    io_manager.check_foreign_keys()
+    check_foreign_keys(engine)
     return 0
 
 
+class ForeignKeyError(sa.exc.SQLAlchemyError):
+    """Raised when data in a database violates a foreign key constraint."""
+
+    def __init__(
+        self, child_table: str, parent_table: str, foreign_key: str, rowids: list[int]
+    ):
+        """Initialize a new ForeignKeyError object.
+
+        Args:
+            child_table: The table that a foreign key constraint is applied to.
+            parent_table: The table that a foreign key constraint refers to.
+            foreign_key: Comma seperated string of key(s) that make up foreign key.
+            rowids: Row(s) of child_table where constraint failed.
```

Comment:
Like the entire rows?  Or the dataframe index of the rows? Or the primary keys of the rows? Or the SQLite autoincrementing integer pseudokey associated with the rows?